src/Monolog/Handler/DeduplicationHandler.php

```diff
@@ -70,13 +70,24 @@ public function flush(): void
             return;
         }
 
+        $gc = false;
+        $store = null;
+
+        if (file_exists($this->deduplicationStore)) {
+            $store = file($this->deduplicationStore, FILE_IGNORE_NEW_LINES | FILE_SKIP_EMPTY_LINES);
+        }
+
         $passthru = null;
 
         foreach ($this->buffer as $record) {
             if ($record->level->value >= $this->deduplicationLevel->value) {
-                $passthru = $passthru === true || !$this->isDuplicate($record);
+                $passthru = $passthru === true || !is_array($store) || !$this->isDuplicate($store, $record);
                 if ($passthru) {
-                    $this->appendRecord($record);
+                    $line = $this->buildDeduplicationStoreEntry($record);
+                    file_put_contents($this->deduplicationStore, $line . "\n", FILE_APPEND);
+                    if (is_array($store)) {
+                        $store[] = $line;
+                    }
```

Comment:
Let's make sure the store becomes active here to avoid sending duplicate lines into it if the buffer contains multiple of the same log records.

```suggestion
                    if (!is_array($store)) {
                        $store = [];
                    }
                    $store[] = $line;
```